test/unittests/test_xmlupload/test_stash_circular_references.py

```diff
@@ -0,0 +1,19 @@
+# pylint: disable=missing-class-docstring,missing-function-docstring,protected-access
+
+import pytest
+
+from dsp_tools.utils.xml_utils import parse_and_clean_xml_file
+from dsp_tools.utils.xmlupload.stash_circular_references import remove_circular_references
+from dsp_tools.utils.xmlupload.xmlupload import _extract_resources_from_xml
+
+test_root = parse_and_clean_xml_file("testdata/generate_test_data/test_circular_references_1.xml")
+resources = _extract_resources_from_xml(test_root, "simcir")
```

Comment:
Is there a reason these 2 are module-level variables? If it is possible, I would rather move them into the method.

Reply:
I can do that, if you prefer it.